File: Scripts/Player/Weapons/Weapon.cs
/*
Generic weapon class that stores basic wepon info
By: Nick Petruccelli
Last Modified: 08/10/2023
*/

using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Weapon : MonoBehaviour
{
    public string wepName;
    public float damage;
    public float critChance;
    public float critMultiplier;
    public GameObject groundItem;
    public float pickUpOffSet;

}

File: Scripts/Player/Weapons/weapon.py
"""
Generic weapon class that stores basic weapon info
By: Nick Petruccelli
Last Modified: 12/26/2023
"""
from dataclasses import dataclass


@dataclass
class Weapon:
    weapon_name: str
    base_damage: float
    crit_chance: float
    crit_mult: float
